### ISSUE TITLE
Bring back pull to refresh on 'Changes' and 'Packages' tab!

### ISSUE THREAD

**Issue by respiredev** [zbrateam/Zebra#1546]
**Is your feature request related to a problem? Please describe.**
Pull to refresh is gone on beta 1.2-4c for the 'Changes' and 'Packages' tab

**Describe the solution you'd like**
Bring back pull to refresh!

**Describe alternatives you've considered**
A refresh button? idk man

**Additional context**
nah

**Comment by aabosh** [comment]
Can you explain why you believe this behaviour should remain? If other people agree, we are open to reconsidering.

**Comment by respiredev** [comment]
I don't see the reasoning for removing it? Is it a performance improvement to get rid of it?
I get the point of automatic refresh, but why not allow people the ability to refresh wherever they are browsing. 

Personally, whenever I open Zebra, I go straight to the Packages tab and pull to refresh there. 
Sometimes when I'm working on a tweak and I install it via theos/SSH, it won't show up until I refresh.

**Comment by aabosh** [comment]
In your situation though, you can just navigate to the Sources tab and refresh from there, no?

For some background, we removed it because the shared code between Packages/Sources/Changes was difficult to maintain and buggy. Rather than rewriting it, which would take time, we felt it would be more valuable to invest that time into other features.

Also, another reason we removed it from other tabs was because you weren't alerted of errors unless you were on the Sources tab.

**Comment by worldlydev** [comment]
Definitely a nice feature that would be nice to have reimplemented. If it's such a hassle to rewrite couldn't the issue be left open until this feature could be focused on again?

**Comment by worldlydev** [comment]
Don't see the point of pull to refresh on the packages tab though. Would much rather have a pull to search. But feel it makes sense on both the sources tab and changes tab.

**Comment by vitalik2187836** [comment]
Don't think we need pull to refresh on packages tab (agreed with the previous comment, pull to search would be better) but it's extremely uncomfortable without this feature on changes tab. Please bring this back.

**Comment by BitGrub** [comment]
Zebra is the only app in the history of iphones that doesn’t let you refresh the main feed FROM the main feed. The removal of sensical refreshing isn’t pointless, it’s actively dumb design

**Comment by wstyres** [comment]
I can see why refresh would be wanted on Changes, it makes sense to have it there and it was removed due to buggy behavior. When we redesign the tab I will add it back to Changes. Packages having refresh doesn't make sense to me because by the same logic you are *just* refreshing the packages that are installed.

**Comment by respiredev** [comment]
My reasoning behind having the pull to refresh on the Packages tab was because in the stable version of Zebra, refreshing on Packages would refresh the sources and any updates would pop up. Whenever I installed tweaks via SSH/Theos or updated my personal repo, I would just refresh Packages and the update would pop up, and it was easy just to click upgrade. 

I understand why people don't think it's necessary. It's really just a QOL thing for certain people I guess. idk

Definitely think It should come back for the Changes tab though! Pull to Search would be cool as well.

**Comment by yur1xpp** [comment]
Yes bring it back please. I just updated to beta from stable version, and this is the only thing I find very "unnatural" about the beta. It doesn't make much sense by not having drag to refresh in "Changes" tab.

@aabosh 
> Personally, whenever I open Zebra, I go straight to the Packages tab and pull to refresh there.
> Sometimes when I'm working on a tweak and I install it via theos/SSH, it won't show up until I refresh.
I couldn't agree more on this.

>In your situation though, you can just navigate to the Sources tab and refresh from there, no?
As mentioned above, when installing tweak via SSH or during tweak development, it's much natural to pull from either "Changes" or "Packages" tab, because I already know I'll be doing stuffs in either of those tabs. Going to "Sources" just for the sake of refreshing and back again to "Changes" or "Packages"  seems very inconvenience design wise.

@wstyres 
>Packages having refresh doesn't make sense to me because by the same logic you are just refreshing the packages that are installed.
Speaking of that, would it be possible to update the installed packages _before the refreshing of sources finished_? I find it very weird that we have to wait until the sources done refreshing for newly installed packaged to show. By new here I means the new package that installed via SSH. Doing tweak development is really a pain when some of the sources downed or slow, making us have to wait for quite some time before we can actually doing anything useful.
